### PR TITLE
e2e test for Kmesh daemon restart

### PR DESCRIPTION
**Pull request by YaoZengzeng**
**What type of PR is this?**

/kind enhancement

**What this PR does / why we need it**:

Add E2E case for that managed applications can work properly before, during and after Kmesh daemon restart

**Which issue(s) this PR fixes**:
Fixes part of #642

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by YaoZengzeng** [comment]
@hzxuzhonghu PTAL

File: test/e2e/main_test.go [hunk 1]
```diff
@@ -176,6 +180,7 @@ func SetupApps(t resource.Context, i istio.Instance, apps *EchoDeployments) erro
 	}
 	apps.All = echos
 	apps.EnrolledToKmesh = match.ServiceName(echo.NamespacedName{Name: EnrolledToKmesh, Namespace: apps.Namespace}).GetMatches(echos)
+	apps.ServiceWithWaypointAtServiceGranularity = match.ServiceName(echo.NamespacedName{Name: ServiceWithWaypointAtServiceGranularity, Namespace: apps.Namespace}).GetMatches(echos)
```
Comment (hzxuzhonghu) [inline test/e2e/main_test.go:183]: I am confused how did you test waypoint-captuired service before without an app

Comment (YaoZengzeng) [reply test/e2e/main_test.go:183]: waypoint-captured service has always existed. We are not going to creat here, but just assign it to this variable for easy use in test case later.

**Comment by YaoZengzeng** [comment]
@hzxuzhonghu updated

File: test/e2e/main_test.go [hunk 2]
```diff
@@ -74,6 +74,9 @@ type EchoDeployments struct {
 	// The echo service which is enrolled to Kmesh without waypoint.
 	EnrolledToKmesh echo.Instances
 
+	// The echo service which is enrolled to Kmesh and with service waypoint.
+	ServiceWithWaypointAtServiceGranularity echo.Instances
+
 	// WaypointProxies by
```
Comment (hzxuzhonghu) [inline test/e2e/main_test.go:80]: nit: by what

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**